src/server_manager/web_app/app.ts

```diff
@@ -727,25 +741,13 @@ export class App {
     }
   }
 
-  // Opens the screen to create a server.
-  private showCreateServer() {
-    const regionPicker = this.appRoot.getAndShowRegionPicker();
-    // The region picker initially shows all options as disabled. Options are enabled by this code,
-    // after checking which regions are available.
-    this.digitalOceanRetry(() => {
-          return this.digitalOceanRepository.getRegionMap();
-        })
-        .then(
-            (map) => {
-              const locations = Object.entries(map).map(([cityId, regionIds]) => {
-                return this.createLocationModel(cityId, regionIds);
-              });
-              regionPicker.locations = locations;
-            },
-            (e) => {
-              console.error(`Failed to get list of available regions: ${e}`);
-              this.appRoot.showError(this.appRoot.localize('error-do-regions'));
-            });
+  private async showCreateServer() {
+    const digitalOceanCreateServer =
+        this.appRoot.getAndShowDigitalOceanCreateServer() as DigitalOceanCreateServer;
+    digitalOceanCreateServer.app = this;
+    digitalOceanCreateServer.digitalOceanRepository = this.digitalOceanRepository;
+    digitalOceanCreateServer.notificationManager = new NotificationManager(this.appRoot);
+    await digitalOceanCreateServer.show();
```

Comment:
Pass the parameters above to `show()` instead. This way we make sure they are always set on show.

Reply:
👍 

I think the cleanest interface would be something like `DigitalOceanCreateServer#show(account: Account)` (where in this case, the `digitalOceanRepository` acts as the account model until we finish the refactor.

It's probably best to register the `NotificationView` when we construct the `DigitalOceanCreateServer`, so I'll use data binding in appRoot... _as long as I can find the right lifecycle methods to use._